### PR TITLE
Add net481 support

### LINKED ISSUE #4224: .Net-Framework 4.8.1 not selectable
### Description

For the targeted framework, you can supply every .Net framework version up to 48, but not 4.8.1. As far as my non-existing F# knowledge tells me, you probably only need to add a few lines in FrameworkHandling.fs.

### Repro steps

In the paket.dependencies, set

framework: net481

which gives an error.

### PR DESCRIPTION
**Pull request by ChrSteinert**
This aims to add support for net481 framework restriction, and thus resolve #4224 .

The required change seems to be more intricate, than what has changed so far. For instance, there is something about the NuGetPackageCache.CurrentCacheVersion, that I am not sure about, what to do. It also touched a lot of moving parts (and tests) that I adjusted straight forward, and might be missing some things.

Therefore any advice is apreciated.

### CODE REVIEW

File: src/Paket.Core/Versioning/FrameworkHandling.fs [hunk 1]
```diff
@@ -919,6 +923,7 @@ type FrameworkIdentifier =
         | DotNetFramework FrameworkVersion.V4_7_1 -> [ DotNetFramework FrameworkVersion.V4_7; DotNetStandard DotNetStandardVersion.V2_0 ]
         | DotNetFramework FrameworkVersion.V4_7_2 -> [ DotNetFramework FrameworkVersion.V4_7_1 ]
         | DotNetFramework FrameworkVersion.V4_8 -> [ DotNetFramework FrameworkVersion.V4_7_2 ]
+        | DotNetFramework FrameworkVersion.V4_8_1 -> [ DotNetFramework FrameworkVersion.V4_7_2 ]
```
Comment (smoothdeveloper) [inline src/Paket.Core/Versioning/FrameworkHandling.fs:926]: I think there is a missing fallback to 4_8 here.

Comment (smoothdeveloper) [reply src/Paket.Core/Versioning/FrameworkHandling.fs:926]: rather, 4_8 should replace 4_7_2 here

**Comment by smoothdeveloper** [comment]
@ChrSteinert could you add one test similar to 
[tests/Paket.Tests/DependenciesFile/ParserSpecs.fs](https://github.com/fsprojects/Paket/pull/4219/files#diff-9c2eeade9b1e471e52ef888ccf82f258a1f39fd27cc78b5f718bab848d621d12)

Otherwise, if CI gives similar output as #4219, I think we can move this forward.

File: src/Paket.Core/Versioning/FrameworkHandling.fs (hunk 1 above)
Comment (ChrSteinert) [reply src/Paket.Core/Versioning/FrameworkHandling.fs]: Changed as requested. Tough I do not fully understand, what that functions intent is…

**Comment by smoothdeveloper** [comment]
@forki, I checked the flakytests CI outputs, they look similar to #4219 AFAICS (beside it pulls dotnet 7 instead of 6).

The added tests cover the support for 4.8.1, I think it is good to go in prerelease mode.

**Review by smoothdeveloper — APPROVED** [review]

**Comment by ChrSteinert** [comment]
Thanks for your guidance and review!
________________________________
Von: Gauthier Segay ***@***.***>
Gesendet: Thursday, September 28, 2023 7:43:04 AM
An: fsprojects/Paket ***@***.***>
Cc: Christian Steinert ***@***.***>; Mention ***@***.***>
Betreff: Re: [fsprojects/Paket] Add net481 support (PR #4227)


@forki<https://github.com/forki>, I checked the flakytests CI outputs, they look similar to #4219<https://github.com/fsprojects/Paket/pull/4219> AFAICS (beside it pulls dotnet 7 instead of 6).

The added tests cover the support for 4.8.1, I think it is good to go in prerelease mode.

**Pull request merged**